src/index.js

```diff
@@ -202,7 +202,7 @@ exports.createFromPrivKey = function (key, callback) {
       return callback(err)
     }
 
-    callback(null, new PeerId(digest, privKey))
+    callback(null, new PeerId(digest, privKey, privKey.public))
```

Comment:
It should check if the public key is on the privKey, what was the case where it missed the public Key?

Reply:
@diasdavid the pubKey check only happens when it's requested, it's undefined before that because it isn't created in the constructor.

This method is used by [`js-ipfs` pre-start](https://github.com/ipfs/js-ipfs/blob/master/src/core/components/pre-start.js#L21). providing pubKey at this point means [the key will also be validated](https://github.com/ya7ya/js-peer-id/blob/c450da00e197337580e7a85c2b4859de845c5144/src/index.js#L16)